tensorflow/python/ops/math_ops.py

```diff
@@ -4857,7 +4857,177 @@ def sparse_segment_sum(
+  array([[[ 1,  2,  3],
+          [ 4,  5,  6]],
+         [[ 7,  8,  9],
+          [10, 11, 12]]])>
+  >>> mat2 = tf.constant(np.arange(13, 25), shape=[2, 3, 2])
+  >>> mat2
+  <tf.Tensor: shape=(2, 3, 2), dtype=int64, numpy=
+  array([[[13, 14],
+          [15, 16],
+          [17, 18]],
+         [[19, 20],
+          [21, 22],
+          [23, 24]]])>
+  >>> tf.sparse.sampled_addmm(indices, values, dense_shape, mat1, mat2, alpha=0.75, beta=0.25)
+  (<tf.Tensor: shape=(2, 2, 2), dtype=int32, numpy=
+  array([[[0, 1],
+          [1, 0]],
+         [[0, 0],
+          [1, 0]]], dtype=int32)>, <tf.Tensor: shape=(2, 2), dtype=float32, numpy=
+  array([[ 75.75, 173.  ],
+         [381.5 , 524.5 ]], dtype=float32)>, 
+  <tf.Tensor: shape=(2,), dtype=int32, numpy=array([2, 2], dtype=int32)>)
+
+  Args:
+    indices: `tf.Tensor` containing coordinates for the rows and columns to be multiplied.
+      Must have rank > 1.
+    values: `tf.Tensor` containing the values to be scaled and added to the sampled dot product.
+    dense_shape: `tf.Tensor` defining the dense shape of the output.
+    mat1: `tf.Tensor` to be multiplied. Must have rank > 1.
+    mat2: `tf.Tensor` to be multiplied. Must have rank > 1.
+    beta: Number to be multipled with `values`. Defaults to 1.0.
+    alpha: Number to be multiplied with the sampled dot product of `mat1` and `mat2`.
+      Defaults to 1.0.
+    output_type: the output datatype if needed. Defaults to float32.
+
+  Returns:
+    Three `tf.Tensor` objects making up the components of a `SparseTensor` representing the result
+    of the operation.
+
+    output = `alpha` * (`mat1` @ `mat2`) * spy(`indices`) + `beta` * `values`
+
+    The function `spy(indices)` is the sparsity pattern matrix derived from `indices`. 
+  """
+  if context.executing_eagerly():
+    if not isinstance(indices, ops.EagerTensor):
+      indices = ops.convert_to_tensor(indices)
+    if not isinstance(values, ops.EagerTensor):
+      values = ops.convert_to_tensor(values)
+    if not isinstance(mat1, ops.EagerTensor):
+      mat1 = ops.convert_to_tensor(mat1)
+    if not isinstance(mat2, ops.EagerTensor):
+      mat2 = ops.convert_to_tensor(mat2)
+  else:
+    if not isinstance(indices, tensor_lib.Tensor):
+      indices = ops.convert_to_tensor(indices)
+    if not isinstance(values, tensor_lib.Tensor):
+      values = ops.convert_to_tensor(values)
+    if not isinstance(mat1, tensor_lib.Tensor):
+      mat1 = ops.convert_to_tensor(mat1)
+    if not isinstance(mat2, tensor_lib.Tensor):
+      mat2 = ops.convert_to_tensor(mat2)
+
+  if values.dtype != output_type:
+    values = cast(values, output_type)
+  if mat1.dtype != output_type:
+    mat1 = cast(mat1, output_type)
+  if mat2.dtype != output_type:
+    mat2 = cast(mat2, output_type)
+
+  dense_rows = mat1.shape[-2]
+  dense_cols = mat2.shape[-1]
+
+  # TODO(mattbahr): use dense_shape to validate the shapes of mat1 and mat2
+  dense_shape = constant_op.constant([dense_rows, dense_cols])
```

Comment:
Having issues accessing the values in dense_shape. Everything I've tried causes my tests to fail in graph mode.

Reply:
Importing `check_ops` introduces a circular dependency. I tried using a regular `assert`, but I kept getting `AssertionErrors` when I run my unit tests.

Reply:
I ended up implementing the check myself and using `gen_logging_ops`. Added a new test case as well and all seems to work ok. If you think this is too much then let me know, and I can remove it. I'm wondering if maybe I'm putting logic in this file that shouldn't be there. For now I'm going to mark this convo resolved.